### PR TITLE
[App] Revert raising error from LightningClient on 500

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Partially reverts #16631.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

File: src/lightning/app/utilities/network.py [hunk 1]
```diff
@@ -139,8 +139,6 @@ def wrapped(*args: Any, **kwargs: Any) -> Any:
             try:
                 return func(self, *args, **kwargs)
             except lightning_cloud.openapi.rest.ApiException as e:
-                if e.status == 500:
-                    raise HTTPException(status_code=500, detail=ast.literal_eval(e.body.decode("utf-8"))["message"])
```
Comment (akihironitta) [inline src/lightning/app/utilities/network.py:143]:
```python
                if e.status == 500 and "some expected error msg" in e:  # or e.message
                    raise HTTPException(status_code=500, detail=ast.literal_eval(e.body.decode("utf-8"))["message"])
```

Comment (tchaton) [reply src/lightning/app/utilities/network.py:143]: It is fine. Let's remove it.

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by awaelchli — APPROVED** [review]
shouldn't there be a test failing when making this change?

**Review by Borda — APPROVED** [review]

**Comment by tchaton** [comment]
> shouldn't there be a test failing when making this change?

Yes, but I commented the tests as the platform API are changing. No problem. This change is fine.

Best,
T.C

**Pull request merged**